ruby/anagram.rb

```diff
@@ -0,0 +1,16 @@
+class Anagram
+  attr_reader :root_word
+
+  def initialize(root_word)
+    @root_word = root_word
+  end
+
+  def match(candidates)
+    anagrams = []
+    for c in candidates do
+      anagrams.push(c) if (c.downcase != self.root_word.downcase) && (c.downcase.split(//).sort == self.root_word.downcase.split(//).sort)
```

Comment:
Can you dry this up. There seems to be a lot of duplication. Also, can it be split in multiple lines to be more readable?